src/librbd/ImageCtx.cc

```diff
@@ -274,6 +274,11 @@ struct C_InvalidateCache : public Context {
     trace_endpoint.copy_name(pname);
     perf_start(pname);
 
+    assert(image_watcher == NULL);
+    image_watcher = new ImageWatcher<>(*this);
+  }
+
+  void ImageCtx::post_init() {
```

Comment:
Nit: perhaps rename to ```init_cache``` since it is for the cache and readahead (which technically requires the cache)

Reply:
@dillaman updated